src/engine_redraw.c

```diff
@@ -518,6 +519,70 @@ void redraw_creature_view(void)
     message_draw();
     gui_draw_all_boxes();
     draw_tooltip();
+    long x = scale_value_by_horizontal_resolution(148);
+    long y = (MyScreenHeight - (scale_value_by_vertical_resolution((LbTextLineHeight())) * 2));
+    int ps_units_per_px;
+    {
+        struct TbSprite* spr = &gui_panel_sprites[488];
+        ps_units_per_px = (22 * units_per_pixel) / spr->SHeight;
+    }
+    for (int Spell = SplK_Freeze; Spell < SplK_TimeBomb; Spell++)
+    {
+        if (creature_affected_by_spell(thing, Spell))
+        {
+            struct SpellInfo* spinfo = get_magic_info(Spell);
+            draw_gui_panel_sprite_left(x, y, ps_units_per_px, spinfo->medsym_sprite_idx);
+            x += scale_value_by_horizontal_resolution(LbTextLineHeight());
+        }
+    }
+    struct CreatureControl* cctrl = creature_control_get_from_thing(thing);
+    if ( (cctrl->dragtng_idx != 0) && ((thing->alloc_flags & TAlF_IsDragged) == 0) )
+    {
+        struct Thing* dragtng = thing_get(cctrl->dragtng_idx);
+        x = MyScreenWidth - (x / 4);
+        unsigned long spr_idx;
+        switch(dragtng->class_id)
+        {
+            case TCls_Object:
+            {
+                struct RoomConfigStats *roomst;
+                if (thing_is_workshop_crate(dragtng))
+                {
+                    roomst = get_room_kind_stats(RoK_WORKSHOP);
+                }
+                else
+                {
+                    roomst = get_room_kind_stats(RoK_LIBRARY);
+                }
+                spr_idx = roomst->medsym_sprite_idx;
+                break;
+            }
+            case TCls_DeadCreature:
+            case TCls_Creature:
+            {
+                spr_idx = get_creature_model_graphics(dragtng->model, CGI_HandSymbol);
+                if (dragtng->class_id == TCls_DeadCreature)
+                {
+                    spr_idx++;
+                }
+                break;
+            }
+            default:
+            {
+                spr_idx = 0;
+                break;
+            }
+        }
+        draw_gui_panel_sprite_left(x, y, ps_units_per_px, spr_idx);
+    }
+    else if (first_person_dig_claim_mode == 1)
+    {
+        if (cctrl->active_instance_id == CrInst_FIRST_PERSON_DIG)
+        {
+            x = MyScreenWidth - (x / 4);
+            draw_gui_panel_sprite_left(x, y, ps_units_per_px, instance_button_init[CrInst_FIRST_PERSON_DIG].symbol_spridx);
+        }
+    }
```

Comment:
instead of doing everything in redraw_creature_view could you just add a new function which just gets called from there instead